tycho-p2-director-plugin/src/main/java/org/eclipse/tycho/plugins/p2/director/DirectorMojo.java

```diff
@@ -378,9 +392,15 @@ public void execute() throws MojoExecutionException, MojoFailureException {
         args.addNonNull("-iuProfileproperties", iuProfileproperties);
         args.addNonNull("-flavor", flavor);
         args.addNonNull("-bundlepool", bundlepool);
-        args.addNonNull("-p2.os", p2os);
-        args.addNonNull("-p2.ws", p2ws);
-        args.addNonNull("-p2.arch", p2arch);
+        if (p2os != null || p2ws != null || p2arch != null) {
+            if (p2os == null || p2ws == null || p2arch == null) {
+                throw new MojoFailureException(
+                        "If any of p2os / p2ws / p2arch is specified, all of them must be specified.");
```

Comment:
can we mention what parameter is missing, e.g. 
```suggestion
         String msg =  "All of p2os / p2ws / p2arch must be specified, ";
         if (p2os!=null) {
            msg+= "p2os="+p2os+ "given";
         } else {
            msg+= "p2os is missing";
         }
         ...
```
That is a bit cumbersome to wirte but users will like good and clear error messages without doubts.